_inc/client/components/navigation-settings/index.jsx

```diff
@@ -78,55 +83,102 @@ export const NavigationSettings = React.createClass( {
 	},
 
 	/**
-	 * The UrlSearch mixin callback to form a new location href string.
+	 * Check that the module list includes at least one of these modules.
 	 *
-	 * @param {string} href the current location string
-	 * @param {string} keyword the new search keyword
-	 * @return {string} href the new location string
+	 * @param  {array}   modules Modules that are probably included in the module list.
+	 *
+	 * @return {boolean}         True if at least one of the modules is included in the list.
 	 */
-	buildUrl: function( href, keyword ) {
-		const splitUrl = href.split( '#' ),
-			splitHash = splitUrl[ 1 ].split( '?' );
-
-		this.props.searchForTerm( keyword );
-		return '#' + splitHash[ 0 ] + ( keyword ? '?term=' + keyword : '' );
```

Comment:
I think removing this broke search.  It refreshes the page and adds params weird places, for example searching for google results in `&s=google#/settings`